docs/source/reference/ha.rst

```diff
@@ -0,0 +1,301 @@
+:orphan:
+
+High availability deployment
+============================
+
+|st2| has been systematically built with High availability(HA) as a goal. The exact deployment
```

Comment:
Maybe we can also some how call out that most components are horizontally scalable and adding more instances of those components increases the overall throughout (well, until RabbitMQ is saturated so right now this is our main bottle neck).